sigmadsp/communication/sigma_tcp_server.py

```diff
@@ -191,23 +104,28 @@ def handle(self):
         """
         while True:
             try:
-                # Receive the packet header in the beginning.
-                packet_header = self.receive_amount(SigmaStudioRequestHandler.HEADER_LENGTH)
+                packet: SigmaProtocolPacket = SigmaProtocolPacket(self.dsp_type)
+                packet.init_from_network(self)
 
-                # The first byte of the header contains the command from SigmaStudio.
-                command = packet_header[0]
+                if packet.header.is_write_request:
+                    self.handle_write_data(packet)
+                elif packet.header.is_read_request:
+                    self.handle_read_request(packet)
 
-                if command == SigmaStudioRequestHandler.COMMAND_WRITE:
-                    self.handle_write_data(packet_header)
+            except ConnectionError:
+                break
 
-                elif command == SigmaStudioRequestHandler.COMMAND_READ:
-                    self.handle_read_request(packet_header)
 
-                else:
-                    logging.info("Received an unknown command code '%s'.", hex(command))
+class Adau145xRequestHandler(SigmaStudioRequestHandler):
+    """ADAU144x/5x/6x request handler."""
 
-            except ConnectionError:
-                break
+    dsp_type = "adau145x"
```

Comment:
This is incompatible with the backend's check for the dsp type, which only accepts the string `adau14xx`. I find it dangerous to compare against dsp_type strings which are defined in different places.

As a fix, this should read:
```suggestion
    dsp_type = "adau14xx"
```

Reply:
it will be misleading, tho - 14xx includes 1401, which is basically automotive-grade 1701

Reply:
ah, I see now, in `sigmadsp.hardware` it's the old version / 14xx; I'll commit these and leave it to the single-source-of-truth changes to address the misleading part